### ISSUE TITLE
Offer definition should mention that businessFunction defaults to "sell"

### ISSUE THREAD

**Issue by danbri** [schemaorg/schemaorg#2357]
Spinning this off from #2348 -

> We should probably also update the definition of Offer to mention the defaulting directly

Currently we state only in businessFunction that 

> The default is http://purl.org/goodrelations/v1#Sell.

...but we don't say very clearly when such a "default" is to be taken into account. 

Setting aside Demand and TypeAndQuantityNode, we should at least amend Offer to mention this.

**Comment by danbri** [comment]
Perhaps something along these lines?

> "The definition of businessFunction notes that its default value is http://purl.org/goodrelations/v1#Sell. Consequently, descriptions of Offers without such a property will typically be intended as offers to sell something. It is best to be explicit and include a businessFunction property for each Offer. Consumers of Offer data who are applying this defaulting heuristic should take care to ensure that businessFunction property values have not been accidentally removed (e.g. in a query), to ensure that any application of a default value reflects the intent of the original publisher.

**Comment by danbri** [comment]
+ should we say anything in https://schema.org/AggregateOffer about homogeneity of the aggregated offers, i.e. do we expect them all to share (explicit or defaulted) businessFunction?

**Comment by RichardWallis** [comment]
One would assume that to be the case, so saying so makes sense.

**Comment by danbri** [comment]
Ping @mfhepp - any thoughts on this?

**Comment by RichardWallis** [comment]
**Suggested updated description for Offer:**

> An offer to transfer some rights to an item or to provide a service — for example, an offer to sell tickets to an event, to rent the DVD of a movie, to stream a TV show over the internet, to repair a motorcycle, or to loan a book.
> 
> Note: As the [[businessFunction]] property, which identifies the form of offer (e.g. sell, lease, repair, dispose), defaults to http://purl.org/goodrelations/v1#Sell; an Offer without a defined businessFunction value can be assumed to be an offer to sell.
> 
> For [GTIN](http://www.gs1.org/barcodes/technical/idkeys/gtin)-related fields, see [Check Digit calculator](http://www.gs1.org/barcodes/support/check_digit_calculator) and [validation guide](http://www.gs1us.org/resources/standards/gtin-validation-guide) from [GS1](http://www.gs1.org/).

**Suggested updated description for AggregateOffer:**

> When a single product is associated with multiple offers (for example, the same pair of shoes is offered by different merchants), then AggregateOffer can be used.
> 
> Note: AggregateOffers are normally expected to associate multiple offers that all share the same defined [[businessFunction]] value, or default to http://purl.org/goodrelations/v1#Sell if businessFunction is not explicitly defined.

**Comment by mfhepp** [comment]
I am fine with the wording that @RichardWallis proposes.

**Comment by danbri** [comment]
@mfhepp thanks. Any thoughts on Demand and TypeAndQuantityNode? does it make sense for them to default? Should we move the "default" language off of the property on to the types?

**Comment by mfhepp** [comment]
@danbri For Demand, we should use a similar wording - keep in mind that there is no "Buy" business function (historically Offer included supply and demand, and the distinction was made solely via the type of the link from the agent to the offer - "offers" for offers and "seeks" for demand. But then I decided that such a fundamental difference should be represented using distinct types and not depend on the type of a link to the offer entity.

For interest in buying something, the business function should be "Sell", for interest in renting, "Lease Out" etc. - The idea is that offer and supply modeling are structurally fully equivalent - very powerful yet underexploited ;-).

For TypeAndQuantityNode, I would also use the same wording. But we need to be careful that if the Offer or Demand entity has a business function, then the default (if we want to continue with defaults) should be the business function given in the main Offer entity (e.,g. if the offer says "LeaseOut", then the assumption for a TypeAndQuantityNode without a business function should also be "LeaseOut", not sell.

Since TypeAndQuantityNode is not yet used too much (unfortunately), we could also state that if you use TypeAndQuantityNode, the business functions must be set explicitly. I agree that defaults are logically problematic; I gave in back then because it was an explicit request from Google back in 2010 or so in order to keep markup as simple as possible.
